### PR TITLE
Weapon select fire

### PR DESCRIPTION
**Pull request by SmallPeenwow**
Have fun

### CODE REVIEW

File: paranomal-game/Assets/Scripts/Weapon/WeaponFiringMode.cs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            if (weapon.isFullAuto)
+            {
+                weapon.isSemiAutomatic = true;
+                weapon.isFullAuto = false;
+            }
+            else
+            {
+                weapon.isSemiAutomatic = false;
+                weapon.isFullAuto = true;
+            }
```
Comment (itbeginshere) [inline paranomal-game/Assets/Scripts/Weapon/WeaponFiringMode.cs:75]: Just look at this section again to see if there isn't a better way to do this, it also looks like the if statements are missing some boolean fields being set to false

**Review by itbeginshere — COMMENTED** [review]
Comment

**Comment by SmallPeenwow** [comment]
image

**Comment by SmallPeenwow** [comment]
Totally forgot you could !variable to change the value of a boolean in C#. Thought only for javascript. I tired my best to make it shorter and hope it takes away you gut pains. @itbeginshere

File: paranomal-game/Assets/Scripts/Weapon/WeaponFiringMode.cs (hunk 1 above)
Comment (SmallPeenwow) [reply paranomal-game/Assets/Scripts/Weapon/WeaponFiringMode.cs]: When testing everything works as planned. Got two guns that use all 2 values. Have a gun that uses full auto and semi auto. Have one that just uses semi. Have one that uses Burst and Semi. Also if wanting to change it please look at the variables allowedSingleShotFire, allowedBurstFire and allowedAutomaticFire as there is a check in start up of application that checks those and when doing the switching of the firing modes

File: paranomal-game/Assets/Scripts/Weapon/WeaponFiringMode.cs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    {
+        // Semi-auto, Burst and Full-auto
+        if(weapon.allowedSingleShotFire && weapon.allowedBurstFire && weapon.allowedAutomaticFire)
+        {
+            SwitchingNotAllowed();
+
+            if (weapon.isFullAuto)
+            {
+                weapon.isSemiAutomatic = true;
+                weapon.isFullAuto = false;
```
Comment (itbeginshere) [inline paranomal-game/Assets/Scripts/Weapon/WeaponFiringMode.cs:34]: what about setting the isBurt to false here?

File: paranomal-game/Assets/Scripts/Weapon/WeaponFiringMode.cs [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+            }
+            else if (!weapon.isFullAuto && weapon.isSemiAutomatic)
+            {
+                weapon.isBurstFire = true;
+                weapon.isSemiAutomatic = false;
```
Comment (itbeginshere) [inline paranomal-game/Assets/Scripts/Weapon/WeaponFiringMode.cs:39]: isAuto = false?

File: paranomal-game/Assets/Scripts/Weapon/WeaponFiringMode.cs [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+            }
+            else if (weapon.isBurstFire)
+            {
+                weapon.isFullAuto = true;
+                weapon.isBurstFire = false;
```
Comment (itbeginshere) [inline paranomal-game/Assets/Scripts/Weapon/WeaponFiringMode.cs:44]: isAuto false?

File: paranomal-game/Assets/Scripts/Weapon/WeaponFiringMode.cs (hunk 2 above)
Comment (SmallPeenwow) [reply paranomal-game/Assets/Scripts/Weapon/WeaponFiringMode.cs:34]: Its already false in the awake. I don't recall every playing a game where the weapon starts on burst

File: paranomal-game/Assets/Scripts/Weapon/WeaponFiringMode.cs (hunk 3 above)
Comment (SmallPeenwow) [reply paranomal-game/Assets/Scripts/Weapon/WeaponFiringMode.cs:39]: Auto is already false

File: paranomal-game/Assets/Scripts/Weapon/WeaponFiringMode.cs (hunk 4 above)
Comment (SmallPeenwow) [reply paranomal-game/Assets/Scripts/Weapon/WeaponFiringMode.cs:44]: So if weapon starts on semi it goes to burst and after goes to full auto from what a recall with when guns have 3 options

**Comment by itbeginshere** [comment]
> 

I had to leave a comment in order to submit 😓

**Comment by itbeginshere** [comment]
I can't drop the gun anymore?

**Comment by itbeginshere** [comment]
This is really cool ✋🏿

**Comment by itbeginshere** [comment]
Code optimization will rather be looked at later down the line

**Pull request merged**